python/doc/examples/reliability_sensitivity/reliability/plot_nais.py

```diff
@@ -0,0 +1,237 @@
+
+# %%
+X = ot.RandomVector(ot.Normal(2))
+
+# %%
+# Create the function :math:`g` from a :class:`~openturns.PythonFunction`:
+
+# %%
+def fourBranch(x):
+    x1 = x[0]
+    x2 = x[1]
+    
+    g1 = 5+0.1*(x1-x2)**2-(x1+x2)/math.sqrt(2)
+    g2 = 5+0.1*(x1-x2)**2+(x1+x2)/math.sqrt(2)
+    g3 = (x1-x2)+9/math.sqrt(2)
+    g4 =(x2-x1)+9/math.sqrt(2)
+    
+    return [min((g1,g2,g3,g4))]
+
+g = ot.PythonFunction(2,1,fourBranch)
+print('function g: ', g)
+
+# %%
+# Draw the function :math:`g` to help to understand the shape of the limit state function:
+
+# %%
+graph = ot.Graph('Four Branch function','x1','x2',True,'topright')
+drawfunction = g.draw([-8]*2,[8]*2,[100]*2)
+graph.add(drawfunction)
+view = View(graph)
+
+
+# %%
+# In order to be able to get the NAIS samples used in the algorithm, it is necessary to transform the :class:`~openturns.PythonFunction` into a :class:`~openturns.MemoizeFunction`:
+
+# %%
+g = ot.MemoizeFunction(g)
+
+# %%
+# Create the output random vector :math:`Y = g(X)`:
+
+# %%
+Y = ot.CompositeRandomVector(g, X)
+
+# %%
+# Create the event :math:`\{ Y = g(X) \leq 0 \}`
+# ----------------------------------------------
+
+# %%
+threshold = 0.0
+myEvent = ot.ThresholdEvent(Y, ot.Less(), threshold)
+
+# %%
+# Evaluate the probability with the NAIS technique
+# ------------------------------------------------
+
+# %%
+rhoQuantile = 0.1
+algo = ot.NAIS(myEvent,rhoQuantile)
+
+# %%
+# Now you can run the algorithm
+
+# %%
+algo.run()
+
+# %%
+result = algo.getResult()
+proba = result.getProbabilityEstimate()
+print('Proba NAIS = ',  proba)
+print('Current coefficient of variation = ',
+      result.getCoefficientOfVariation())
+
+# %%
+# The length of the confidence interval of level :math:`95\%` is:
+
+# %%
+length95 = result.getConfidenceLength()
+print('Confidence length (0.95) = ', result.getConfidenceLength())
+
+# %%
+# which enables to build the confidence interval:
+
+# %%
+print('Confidence interval (0.95) = [', proba -
+      length95/2, ', ', proba + length95/2, ']')
+
+# %%
+# Draw the NAIS samples used by the algorithm
+# -------------------------------------------
+#
+# The following manipulations are possible only if you have created a :class:`~openturns.MemoizeFunction` that enables to store all the inputs and outputs of the function :math:`g`.
+#
+# Get all the inputs and outputs where :math:`g` were evaluated:
+
+# %%
+inputNAIS = g.getInputHistory()
+outputNAIS = g.getOutputHistory()
+nTotal = inputNAIS.getSize()
+print('Number of evaluations of g = ', nTotal)
+
+# %%
+# Within each step of the algorithm, a sample of size :math:`N` is created, where:
+
+# %%
+N = algo.getMaximumOuterSampling()*algo.getBlockSize()
+print('Size of each subset = ', N)
+
+# %%
+# You can get the number :math:`Ns` of steps with:
+
+# %%
+Ns =  int(nTotal / N)
+print('Number of steps = ', Ns)
+
+# %%
+# Now, we can split the initial sample into :class:`~openturns.NAIS` samples of size :math:`Ns`:
```

Comment:
```suggestion
# Now, we can split the initial sample into NAIS samples of size :math:`N_s`:
```
If you want to link to the NAIS class, it should be done earlier, before the class is actually constructed.